src/psyclone/tests/domain/lfric/lfric_extract_driver_creator_test.py

```diff
@@ -35,68 +35,48 @@
 
 ''' This module tests the driver creation for extracted kernels.'''
 
+import os
 import pytest
 
 from psyclone.core import Signature
-from psyclone.domain.lfric import LFRicConstants, LFRicExtractDriverCreator
+from psyclone.domain.lfric import LFRicExtractDriverCreator
 from psyclone.domain.lfric.transformations import LFRicExtractTrans
 from psyclone.errors import GenerationError, InternalError
+from psyclone.parse import ModuleManager
 from psyclone.psyir.nodes import Literal, Routine, Schedule
 from psyclone.psyir.symbols import INTEGER_TYPE
 from psyclone.psyir.tools.dependency_tools import DependencyTools
-from psyclone.tests.utilities import get_invoke
+from psyclone.tests.utilities import Compile, get_base_path, get_invoke
 
 
 API = "dynamo0.3"
 
 
-def test_lfric_driver_constructor():
-    '''Tests the constructor of the LFRic driver creator.'''
-
-    driver_creator = LFRicExtractDriverCreator()
-    default_precision = {"i_def": "int32", "r_def": "real64",
-                         "r_second": "real64", "r_solver": "real32",
-                         "r_tran": "real32"}
-
-    assert driver_creator._precision == default_precision
-
-    # Check that we can modify the default precisions:
-    new_precision = {"r_solver": "real64", "r_tran": "real64"}
-    driver_creator = LFRicExtractDriverCreator(new_precision)
-    assert (driver_creator._precision ==
-            {"i_def": "int32", "r_def": "real64",
-             "r_second": "real64", "r_solver": "real64",
-             "r_tran": "real64"})
-
+@pytest.fixture(scope='function')
```

Comment:
Well, only half the tests need it (ok, 9 out of 16). I thought it's easier this way? I don't want to do all this stuff unnecessary for all files, esp. since tests might (at some stage, I don't think atm admittedly) need an empty module manager. Again, if you prefer autouse, I am happy to add.

Reply:
Oh, OK. Presumably I thought the proportion was higher than that. (At least, I'm going to assume that's what I thought ;-) )